pyquil/paulis.py

```diff
@@ -773,7 +781,7 @@ def _exponentiate_general_case(pauli_term, param):
     def reverse_hack(p):
         # A hack to produce a *temporary* program which reverses p.
         revp = Program()
-        revp.inst(list(reversed(p.instructions)))
```

Comment:
Why is this accessing the internal _instructions?

Reply:
at the beginning of this PRs life, the difference was `_instructions` kept the placeholders and `instructions` would address them, so this was needed for exponentiate to actually work. I will reset it